web/packages/teleterm/src/ui/DocumentAccessRequests/NewRequest/useNewRequest.ts

```diff
@@ -349,6 +339,8 @@ function getDefaultSort(kind: ResourceKind): SortType {
   return { fieldName: 'name', dir: 'ASC' };
 }
 
-export type ResourceKind = ResourceIdKind | 'role';
+export type ResourceKind =
+  | Exclude<ResourceIdKind, 'user_group' | 'windows_desktop'>
```

Comment:
I was going to say that this is technically less future-proof, because with every added kind, TypeScript will automatically assume that Connect supports it. So it could be better to explicitly say which ones we support instead of which ones are not supported.

OTOH if there's a place in the code that does an exhaustive check on this type, then adding a new kind will automatically force people to either implement support in Connect or explicitly add it to the ignore list here.

On the other other hand, I don't think there's such place with an exhaustive check in the project at the moment.

But overall it should be fine to leave it like that at the moment.

---

Update: It's not really true though, I just tried adding a new kind. There's a bunch of errors because there are other places in the code which explicitly list which kinds Connect supports.

Reply:
Actually, I like the version with explicit kinds more :)
Changed.